packages/fetchai/connections/webhook/connection.py

```diff
@@ -0,0 +1,267 @@
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""Webhook connection and channel"""
+
+import asyncio
+import json
+import logging
+from asyncio import CancelledError
+from typing import Optional, Union, cast
+
+from aiohttp import web  # type: ignore
+
+from aea.configurations.base import ConnectionConfig, PublicId
+from aea.connections.base import Connection
+from aea.mail.base import Address, Envelope, EnvelopeContext
+
+from packages.fetchai.protocols.http.message import HttpMessage
+from packages.fetchai.protocols.http.serialization import HttpSerializer
+
+SUCCESS = 200
+NOT_FOUND = 404
+REQUEST_TIMEOUT = 408
+SERVER_ERROR = 500
+
+logger = logging.getLogger("aea.packages.fetchai.connections.webhook")
+
+RequestId = str
+
+
+class WebhookChannel:
+    """A wrapper for a Webhook."""
+
+    def __init__(
+        self,
+        agent_address: Address,
+        webhook_address: Address,
+        webhook_port: int,
+        webhook_url_path: str,
+        connection_id: PublicId,
+    ):
+        """
+        Initialize a webhook channel.
+
+        :param agent_address: the address of the agent
+        :param webhook_address: webhook hostname / IP address
+        :param webhook_port: webhook port number
+        :param connection_id: the connection id
+        """
+        self.agent_address = agent_address
+
+        self.webhook_address = webhook_address
+        self.webhook_port = webhook_port
+        self.webhook_url_path = webhook_url_path
+
+        self.webhook_site = None  # type: Optional[web.TCPSite]
+        self.runner = None  # type: Optional[web.AppRunner]
+        self.app = None  # type: Optional[web.Application]
+
+        self.is_stopped = True
+
+        self.connection_id = connection_id
+        self.in_queue = None  # type: Optional[asyncio.Queue]  # pragma: no cover
+        self.loop = (
+            None
+        )  # type: Optional[asyncio.AbstractEventLoop]  # pragma: no cover
+        logger.info("Initialised a webhook channel")
+
+    async def connect(self):
+        """
+        Connect the webhook
+
+        Connects the webhook via the webhook_address and webhook_port parameters
+        """
+        if self.is_stopped:
+            self.app = web.Application()
+            self.app.add_routes(
+                [web.post(self.webhook_url_path, self._receive_webhook)]
+            )
+            self.runner = web.AppRunner(self.app)
+            await self.runner.setup()
+            self.webhook_site = web.TCPSite(
+                self.runner, self.webhook_address, self.webhook_port
+            )
+            await self.webhook_site.start()
```

Comment:
Is this blocking? I hope not but hard to tell without looking at `aiohttp`

Reply:
Does not seem to be blocking. Looking at aiohttp, the `start` method is async and has its own loop. Also from examining the Aries code it, the call does not block.